### ISSUE TITLE
[INFRA-1973] Migrate rating.jenkins.io to Azure

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkins-infra/helpdesk#1627]
<p>This machine is running in a deprecated AWS environment and should be moved as soon as possible.<br/>
This service is already dockerize and should be easy to move on our k8s cluster</p>

---
<details><summary><i>Originally reported by <a title="Olivier Vernin" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=olblak">olblak</a>, imported from: <a href="https://issues.jenkins.io/browse/INFRA-1973" target="_blank">Migrate rating.jenkins.io to Azure</a></i></summary>
<i><ul>
<li><b>status</b>: Open
<li><b>priority</b>: Minor
<li><b>resolution</b>: Unresolved
<li><b>imported</b>: 2022/01/10
</ul></i>
</details>

**Comment by dduportal** [comment]
I believer that @jplayout is interested in this one :)

**Comment by dduportal** [comment]
- Current infra as code definition is there: https://github.com/jenkins-infra/jenkins-infra/blob/production/dist/profile/manifests/rating.pp

**Comment by lemeurherve** [comment]
@jplayout feel free to ping me if you need :)

**Comment by dduportal** [comment]
As rating.jenkins.io need a database, I propose to keep the for the first version (might be slow)

**Comment by dduportal** [comment]
For information, Gaving shared this link with us in the Discourse forum: https://github.com/jenkins-infra/kubernetes-management/pull/137/files

**Comment by dduportal** [comment]
Blocked by #2259 (to allow adding the new azure-managed database)

**Comment by timja** [comment]
I think that's the wrong reference?

**Comment by dduportal** [comment]
> I think that's the wrong reference?

Yep, my bad. Fixed.

**Comment by dduportal** [comment]
Next step: create a terraform-managed database in Azure for the service in AKS

- Azure Terraform repository: https://github.com/jenkins-infra/azure
- Code of the ratings.jenkins.io application: https://github.com/jenkins-infra/rating (Docker application)
- Azure PostgreSQL managed databases: https://azure.microsoft.com/en-us/services/postgresql/
- Terraform Azure PostgreSQL resource: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/postgresql_server

**Comment by timja** [comment]
Probably this instead?
https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/postgresql_flexible_server

Microsoft is moving away from the original one

**Comment by dduportal** [comment]
> Probably this instead? https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/postgresql_flexible_server
> 
> Microsoft is moving away from the original one

Thanks for the tip Tim!

Ref. https://docs.microsoft.com/en-us/azure/postgresql/flexible-server/overview#frequently-asked-questions

**Comment by timja** [comment]
every time we get an outage during maintenance it's 'switch to flexible server'

but we can't due to missing Azure AD support. but fine for here :)

**Comment by dduportal** [comment]
@timja we created the initial postgres server to support this DB (and eventually others). Could you give us a review in https://github.com/jenkins-infra/azure/pull/178 before we proceed to the (create db +user for rating) ?

**Comment by dduportal** [comment]
As per https://github.com/jenkins-infra/helpdesk/issues/1692 (closed as duplicate of this one), we will have to delete the 2 AWS RDS databases once migration is finished.

**Comment by lemeurherve** [comment]
From another issue,
> To dump the original database:
> - ssh rating
> - dump from the machine

**Comment by smerle33** [comment]
- move management to kubernetes : https://github.com/jenkins-infra/kubernetes-management/pull/2137
- backup old bdd from the old VM

**Comment by dduportal** [comment]
- Opened a notice in status.jenkins.io 24h before: https://github.com/jenkins-infra/status/pull/152
- Set the DNS TTL to 60s, 24h before ✅ 
- Migrated TTL the 3rd of May ✅ 
- Checked that new service takes over and that jenkins.io allows voting ✅ 
- Closed notice in status.jenkins.io: https://github.com/jenkins-infra/status/pull/154
- Stopped and disabled service on the AWS VM ✅ 

TODO list for cleanup:
- [ ] Cleanup code related to rating.jenkins.io in Puppet code
- [ ] Delete the AWS VM and its associated resource (data sources)
- [ ] Delete the AWS database and its replica

**Comment by smerle33** [comment]
For Cleanup : 
- [ ] not sure: https://github.com/jenkins-infra/jenkins-infra/blob/f5e81c51648ad933aea5136aa82a0c3ef54a93b2/dist/profile/manifests/catchall.pp#L66
- [ ] https://github.com/jenkins-infra/jenkins-infra/blob/f5e81c51648ad933aea5136aa82a0c3ef54a93b2/dist/profile/manifests/rating.pp
- [ ] https://github.com/jenkins-infra/jenkins-infra/blob/f5e81c51648ad933aea5136aa82a0c3ef54a93b2/dist/role/manifests/rating.pp
- [ ] https://github.com/jenkins-infra/jenkins-infra/blob/f5e81c51648ad933aea5136aa82a0c3ef54a93b2/manifests/site.pp#L109-L112
- [ ] https://github.com/jenkins-infra/jenkins-infra/blob/f5e81c51648ad933aea5136aa82a0c3ef54a93b2/hieradata/common.yaml#L158-L160
- [ ] https://github.com/jenkins-infra/jenkins-infra/blob/f5e81c51648ad933aea5136aa82a0c3ef54a93b2/spec/classes/profile/rating_spec.rb
- [ ] not sure: https://github.com/jenkins-infra/jenkins-infra/blob/f5e81c51648ad933aea5136aa82a0c3ef54a93b2/spec/server/vhostcatchall/vhostcatchall_spec.rb#L29-L32
- [ ] https://github.com/jenkins-infra/jenkins-infra/blob/f5e81c51648ad933aea5136aa82a0c3ef54a93b2/updatecli/weekly.d/ratings.yaml 

WDYT ?

**Comment by dduportal** [comment]
@smerle33 thanks for listing and asking there!

As a rule of thumb, cleaning up the puppet code for rating.jenkins.io means:

- Removing the host from puppet management in https://github.com/jenkins-infra/jenkins-infra/blob/production/manifests/site.pp
- Removing the role associated to this machine if not used anywhere (here: `role::rating`) - https://github.com/jenkins-infra/jenkins-infra/blob/production/dist/role/manifests/rating.pp
- Removing the profiles associated to this role if not used anywhere (`profile::rating` to be removed, but keep `profile::base` because it's used by all the others roles) - https://github.com/jenkins-infra/jenkins-infra/blob/production/dist/profile/manifests/rating.pp
  - Don't forget to also all the templates/files/tests associated to this profile in https://github.com/jenkins-infra/jenkins-infra/blob/production/dist/profile/**
-  Remove all the configuration associated to rating (and not used anywhere else) in hieradatas (both local vagrant and production)
- Remove all the general documentation/tests associated to rating


=> I let you put each of your link in one of these categories to help us validate that it's ok to remove it or not. Most of them are good cactches, a few should be kept or discussed.
=> As seen together, we do not touch the datadog monitoring probes in this change

**Comment by smerle33** [comment]
New version : 
As a rule of thumb, cleaning up the puppet code for rating.jenkins.io means:

- Removing the host from puppet management in https://github.com/jenkins-infra/jenkins-infra/blob/production/manifests/site.pp
- Removing the role associated to this machine if not used anywhere (here: `role::rating`) - https://github.com/jenkins-infra/jenkins-infra/blob/production/dist/role/manifests/rating.pp
- Removing the profiles associated to this role if not used anywhere (`profile::rating` to be removed, but keep `profile::base` because it's used by all the others roles) - https://github.com/jenkins-infra/jenkins-infra/blob/production/dist/profile/manifests/rating.pp
  - Don't forget to also all the templates/files/tests associated to this profile in https://github.com/jenkins-infra/jenkins-infra/blob/production/dist/profile/**
-  Remove all the configuration associated to rating (and not used anywhere else) in hieradatas (both local vagrant and production)
    https://github.com/jenkins-infra/jenkins-infra/blob/f5e81c51648ad933aea5136aa82a0c3ef54a93b2/hieradata/common.yaml#L158-L160
- Remove all the general documentation/tests associated to rating
    https://github.com/jenkins-infra/jenkins-infra/blob/f5e81c51648ad933aea5136aa82a0c3ef54a93b2/updatecli/weekly.d/ratings.yaml
    https://github.com/jenkins-infra/jenkins-infra/blob/f5e81c51648ad933aea5136aa82a0c3ef54a93b2/spec/classes/profile/rating_spec.rb


will be remove on another PR for cleaning catchall profile : 
  -  https://github.com/jenkins-infra/jenkins-infra/blob/f5e81c51648ad933aea5136aa82a0c3ef54a93b2/dist/profile/manifests/catchall.pp#L66
  -  https://github.com/jenkins-infra/jenkins-infra/blob/f5e81c51648ad933aea5136aa82a0c3ef54a93b2/spec/server/vhostcatchall/vhostcatchall_spec.rb#L29-L32

**Comment by dduportal** [comment]
Heads up before week end.

Remaining task to close:

- [ ] Update docs (runbook, readmes, jenkins.io webpage)
- [ ] Delete the managed database in AWS

**Comment by dduportal** [comment]
This issue is closed as we finished the cleanup today!
